resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -59,7 +61,24 @@ object ExecutorPodsSnapshot extends Logging {
         case "pending" =>
           PodPending(pod)
         case "running" =>
-          PodRunning(pod)
+          // TODO(SPARK-29023): Kubernenetes 1.17 sidecar container feature will
+          // make this code redundant https://github.com/kubernetes/enhancements/issues/753
+          // Checking executor container status is not terminated
+          // Pod status can still be running if sidecar container status is running
```

Comment:
What if the side car thing provides a service that we need eg. kerberos renewal, shouldnt we remove the executor pod asap and not wait for the Spark container error to happen? We should emit a log msg at least eg. "Spark executor is running but pod is not healthy". Or we could have a configurable property to allow removal if any container is failing. In addition, Pod restartPolicy is set to `Never` since spark manages errors and since we bypass the K8s deployment pattern we should provide a few more managements capabilities. Thoughts?

Reply:
That's a good point. I don't know how often the sidecar is "critical" to keep running along side Spark so I'll defer to other's judgement here.

Reply:
The current state is if any container is running, the pod will keep running, so there is no case "Spark executor is running but pod is not healthy" but will happen "pod is healthy but Spark executor container(or any sidecar container) is not running "
My change made "main(executor)" container reflect on the pod status.
Your point is very good, that other sidecar also should be healthy when they should be. My thoughts on it are, sidecars are not as critical as "main" container in some case, and should not necessary cause pod to restart all the time. so I will like it to happen is to let "main(executor)" container determine if a pod should be restarted. which should be main container error out when any of its sidecar container is not healthy. 
Let me know what you guys think. thanks.